### DISCUSSION TITLE
Configs: Consider separating out a 'stylistic' config from recommended and strict

### DISCUSSION THREAD

**Discussion by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#6009] [RFCs]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22preset+config+change%22+) and found none that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Description

Forking discussion from @bmish in https://github.com/eslint/eslint/discussions/16512#discussioncomment-4107395:

> What might make more sense is having the recommended focus on bugs, optionally a strict config that is stricter/more-opinionated in someway, and a separate stylistic (or similar name) config focus on formatting. 

I like this. Thoughts?

### Impacted Configurations

* `strict`

**Comment by bradzacher** [comment]
What would be the stylistic rules that we have in recommended? The first recommended config was pretty stylistically opinionated, but the current version is pretty much just bugs I think.

**Comment by JoshuaKGoldberg** [comment]
I think it's now just `strict` that has those, yeah. Edited.
